### ISSUE TITLE
Altitude

### ISSUE THREAD

**Issue by eq2kazial** [iam4x/pokemongo-webspoof#354]
With the discovery that the game also uses altitude in its communications with Niantic's servers, how does this app handle altitude? Is it the same number constantly (and thus easy for them to detect?) or is it something we can customize?

**Comment by jamesjia94** [comment]
The current app only uses Google's Directions API, which does not send elevation information. I'm guessing your phone will continue to send the altitude you were at prior to having used GPS spoofing. One fix could be using Google's elevation API to get elevation data.

**Comment by eq2kazial** [comment]
Using the elevation API would be great. Sadly I'm getting the feeling this app is dead.

**Comment by HoDaDor** [comment]
=(

**Comment by eq2kazial** [comment]
Can @iam4x or any other coder implement altitude to the app?

**Comment by m4hx** [comment]
I am afraid this project is dead. I did some testing with the code and altitude is always 0 when using this gps spoof technique. Unfortunately this relies on using xcode to simulate the GPS coordinates and at this time xcode does not have the ability to simulate altitude.

**Comment by HoDaDor** [comment]
༼ つ ಥ_ಥ ༽つ 

Thanks for the attempt @m4hx

**Comment by MoOx** [comment]
Red flag indeed :/

**Comment by ehad16** [comment]
@m4hx would these links help?

http://stackoverflow.com/questions/9439495/when-using-gpx-in-xcode-to-simulate-location-changes-is-there-a-way-to-control/9439580#9439580

https://developer.apple.com/library/ios/documentation/ToolsLanguages/Reference/UIATargetClassReference/index.html#//apple_ref/javascript/instm/UIATarget/setLocationWithOptions

**Comment by HoDaDor** [comment]
Perhaps set a config variable to set and program in some fluctuation so the elevation variable doesn't remain constant? 

Elevation data here: http://viewer.nationalmap.gov/basic/?basemap=b1&category=ned,nedsrc&title=3DEP%20View

**Comment by reactortrip** [comment]
This looks like it could help, but I don't know how to implement it.

https://github.com/vermont42/GpxLocationManager/blob/master/README.md

**Comment by ehad16** [comment]
This looks like it could help, but I don't know how to implement it.

https://github.com/vermont42/GpxLocationManager/blob/master/README.md

**Comment by nickg819** [comment]
Can you link to where this discovery was found?

**Comment by ehad16** [comment]
Few threads talking about it in the PoGo Dev reddit, here is one.

https://www.reddit.com/r/pokemongodev/comments/4wdpue/so_waitthe_game_verifies_altitude_thats_going_to/

**Comment by filipechagas** [comment]
I've found this guy on the look for alternatives: https://github.com/vermont42/GpxLocationManager

Not sure it helps on this project, and I'm not a ios/swift developer as well, but looks like this guy offers some classes that allow simulating location with elevation information based on gpx files.

**Comment by MoOx** [comment]
It would be nice if someone is comfortable enough with ios/xcode dev to handle this via PR. Should be a high priority since we could all be banned because of this.

**Comment by MoOx** [comment]
I guess if niantic is becoming serious about this (that's what we can read if you take a look to niantic ceo replies on twitter), we could even be banned retroactively...

**Comment by filipechagas** [comment]
You're right. And, besides the elevation (altitude) thing, if they start checking the accelerometer to make sure the player is in motion, we might as well just wind up not being able to use spoofing altogether.

**Comment by eq2kazial** [comment]
Any updates on adding altitude? If not, anyone know another similar program to spoof on iOS that does?

**Comment by MoOx** [comment]
@iam4x is probably on vacation according to his github activity stream. Nobody has access to the repo except himself so we have to wait.
Meanwhile anyone can make a PR with this improvements and everyone will be able to use it.

**Comment by grinsekatze003** [comment]
I doubt there is any solution. The only way to simulate any iOS data without jailbreak is to use Apples tools. But Xcode has afaik no built-in solution to fake altitude. There -was- a way to simulate altitude using automations but I think the support got dropped with iOS10.

**Comment by AnySkill** [comment]
Did actually someone got banned using this ?

**Comment by filipechagas** [comment]
I got a soft ban after using it once.

**Comment by iam4x** [comment]
@AnySkill Not yet for myself.

**Comment by MoOx** [comment]
I just got some soft bans but it was because I moved to quickly (and sometimes, not on purpose).
No permanent ban (yet?) and I used it probably some hours already.

**Comment by adrianoresende** [comment]
@filipechagas I never used other applications and scripts third-party?

**Comment by KTachyon** [comment]
@filipechagas The GpxLocationManager won't work. It's scoped to the app running it, not system wide.

This could be possible to override using Automation, but I don't know if it's feasible.

But there is an alternative... the hardware alternative. On iOS you can connect an external GPS and the system will use it as it's location hardware (at least on iPods and SIM-less iPads, which don't have the GPS chip, but will probably work on every iOS device). Better yet, it seems to be possible to do that over bluetooth.

(And I'm going to stop here and let someone with iOS Bluetooth communication background to take over)

**Comment by imnotabunai** [comment]
Automation doesn't work with the current iteration, I've just checked using various options.

```
var target = UIATarget.localTarget();

var location = {
latitude: 0,
longitude: 0`
};

var locationOptions = {
altitude: 20,
horizontalAccuracy: 10,
verticalAccuracy:15
};

target.setLocationWithOptions (location, locationOptions);
```

Going to keep trying, but I've read around it's not possible anymore (though I can't remember where I read it).

**Comment by GForceWeb** [comment]
Has there been any development on implementing some sort of altitude value into this project?
